src/components/Tooltip/GenericTooltip.tsx

```diff
@@ -79,21 +78,27 @@ function GenericTooltip({
         setIsRendered(true);
         setIsVisible(true);
 
-        animation.current.stopAnimation();
+        cancelAnimation(animationSharedValue);
 
         // When TooltipSense is active, immediately show the tooltip
         if (TooltipSense.isActive() && !shouldForceAnimate) {
-            animation.current.setValue(1);
+            animationSharedValue.value = 1;
         } else {
-            isTooltipSenseInitiator.current = true;
-            Animated.timing(animation.current, {
-                toValue: 1,
-                duration: 140,
-                delay: 500,
-                useNativeDriver: false,
-            }).start(({finished}) => {
-                isAnimationCanceled.current = !finished;
-            });
+            isTooltipSenseInitiatorShared.value = true;
+            animationSharedValue.value = withDelay(
+                500,
+                withTiming(
+                    1,
+                    {
+                        duration: 140,
+                    },
+                    (finished) => {
+                        runOnJS(() => {
+                            isAnimationCanceledShared.value = !finished;
+                        })();
```

Comment:
I'm wondering why do we need to run it on JS thread?

Reply:
I will correct it in that case if it is not needed :)